src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,79 @@
-export const Pagination = () => {};
+import cn from 'classnames';
+import { getNumbers } from '../../utils';
+
+type PropsPagination = {
+  total: number;
+  perPage: number;
+  currentPage: number;
+  onPageChange: (page: number) => void;
+};
+
+export const Pagination = ({
+  total,
+  perPage,
+  currentPage = 1,
+  onPageChange,
+}: PropsPagination) => {
+  const items = getNumbers(1, total);
+  const totalPages = Math.ceil(items.length / perPage);
+  const totalPagesArr = getNumbers(1, totalPages);
+
+  return (
+    <ul className="pagination">
+      <li
+        className={cn(
+          'page-item',
+          { disabled: currentPage === 1 },
+          { active: currentPage === 1 },
+        )}
+      >
+        <a
+          data-cy="prevLink"
+          className="page-link"
+          href="#prev"
+          aria-disabled={currentPage === 1}
+          onClick={() => {
+            if (currentPage !== 1) {
+              onPageChange(currentPage - 1);
+            }
+          }}
+        >
+          «
+        </a>
+      </li>
+      {totalPagesArr.map(p => (
+        <li className={cn('page-item', { active: currentPage === p })} key={p}>
+          <a
+            data-cy="pageLink"
+            className="page-link"
+            href={`#${p}`}
+            onClick={() => onPageChange(p)}
```

Comment:
Using an anchor tag with `href` that is a hash can lead to changes in the URL which might not be desired. Consider using a `button` element or preventing the default action in the click event.

Reply:
I have said this before: if mentors have written initial state of the task with these HTML elements I'm not gonna change it